frontend/test/pytest/test_peephole_optimizations.py

```diff
@@ -191,6 +191,40 @@ def test_chained_apply_passes_workflow(x: float):
 
 test_chained_passes()
 
+#
+# to_ppr
+#
+
+
+def test_convert_clifford_to_ppr():
+    """
+    Test convert_clifford_to_ppr
+    """
+    pipe = [("pipe", ["enforce-runtime-invariants-pipeline"])]
+
+    @qjit(pipelines=pipe, target="mlir")
+    def test_convert_clifford_to_ppr_workflow():
+
+        @to_ppr
+        @qml.qnode(qml.device("lightning.qubit", wires=2))
+        def f():
+            qml.H(0)
+            qml.S(1)
+            qml.T(0)
+            qml.CNOT([0, 1])
+
+        return f()
+
+    assert 'transform.apply_registered_pass "to_ppr"' in test_convert_clifford_to_ppr_workflow.mlir
+    assert (
+        'transform.apply_registered_pass "to_ppr"'
+        not in test_convert_clifford_to_ppr_workflow.mlir_opt
+    )
+    assert "qec.ppr" in test_convert_clifford_to_ppr_workflow.mlir_opt
```

Comment:
Worth noting that each of these asserts runs the compiler (3 in total). We didn't deem it that important to cache the IR in the PR that updated the `.mlir` property, because in an interactive session it probably wouldn't be that relevant, but I now think in tests this could add up if we're not careful. I guess this is one downside of making it a property instead of a function, it's not clear each "invocation" does something. Anyway, just something to keep in mind :)

```suggestion
    optimized_ir = test_convert_clifford_to_ppr_workflow.mlir_opt
    assert 'transform.apply_registered_pass "to_ppr"' not in optimized_ir
    assert "qec.ppr" in optimized_ir
```

Reply:
hmm that's a good observation. I agree that it is easy to forget each .mlir will trigger a new compilation when it is a property. Wondering how difficult it is to cache it 🤔